pandas/tests/generic/test_frame.py

```diff
@@ -47,18 +47,23 @@ def test_set_axis_name_mi(self, func):
 
     def test_nonzero_single_element(self):
         # allow single item via bool method
-        df = DataFrame([[True]])
-        assert df.bool()
-
-        df = DataFrame([[False]])
-        assert not df.bool()
-
-        df = DataFrame([[False, False]])
-        msg = "The truth value of a DataFrame is ambiguous"
-        with pytest.raises(ValueError, match=msg):
-            df.bool()
-        with pytest.raises(ValueError, match=msg):
-            bool(df)
+        msg = (
+            "NDFrame.bool is now deprecated and will be removed in future releases "
+            "and cases that relied on it will raise in a future version"
+        )
+        with tm.assert_produces_warning(FutureWarning, match=msg):
```

Comment:
Each of the calls below should produce warnings. By having a single `with` block you're only testing that _at least one_ of them produces a warning. Instead, you should have a `with ...:` for each call to the deprecated function, as narrowly scoped as possible.

Reply:
done